### PR TITLE
Task A

### PR DESCRIPTION
**Pull request by abhinavachanta**
Please review.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#CREATE BIN AND BUILD FOLDERS TO SAVE THE COMPILED FILES DURING RUNTIME
+bin_folder := $(shell mkdir -p bin)
+build_folder := $(shell mkdir -p build/src build/test/subnet build/test/sender build/test/receiver)
+
+#TARGET TO RUN ONLY BIT SIMULATOR
+simulator: main.o ../data_structures/message.o
+	$(CC) -g -o bin/ABP build/src/main.o build/src/message.o
+clean_simulator:
+	rm -f bin/ABP.exe build/src/*.o *~
+	-for d in src/data_structures; do (cd $$d; rm -f *.o *~ ); done
```
Comment (syedomarazam) [inline makefile]: you can remove this line because we aren't generating .o files in data_structures folder anymore.

File: makefile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#TARGET TO RUN ALL THE TESTS TOGETHER (NOT SIMULATOR)
+	
+tests: test_subnet_main.o test_sender_main.o test_receiver_main.o ../data_structures/message.o
+		$(CC) -g -o bin/SUBNET_TEST build/test/subnet/main.o build/src/message.o
+		$(CC) -g -o bin/SENDER_TEST build/test/sender/main.o build/src/message.o 
+		$(CC) -g -o bin/RECEIVER_TEST build/test/receiver/main.o build/src/message.o  
+		
+clean_tests:
+	rm -f bin/*.exe build/test/**/*.o *~
+	-for d in src/data_structures; do (cd $$d; rm -f *.o *~ ); done
```
Comment (syedomarazam) [inline makefile]: you can remove this line because we aren't generating .o files in data_structures folder anymore.

File: makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+CFLAGS=-std=c++17
+
+INCLUDECADMIUM=-I lib/cadmium/include
+
+#CREATE BIN AND BUILD FOLDERS TO SAVE THE COMPILED FILES DURING RUNTIME
+bin_folder := $(shell mkdir -p bin)
+build_folder := $(shell mkdir -p build/src build/test/subnet build/test/sender build/test/receiver)
+
+#TARGET TO RUN ONLY BIT SIMULATOR
+simulator: main.o ../data_structures/message.o
```
Comment (syedomarazam) [inline makefile]: you can rename "../data_structures/message.o" as it created confusion in the reader's mind. a user might think he has to navigate one directory back and go to data structures folder. this path actually doesn't exist. if you modify it here, make sure you change it in all the places in makefile

**Review by syedomarazam — COMMENTED** [review]
in makefile:

you could probably rearrange code to give more readability to users. something like this
--> creating bin and build folders
--> all:
--> simulator:
--> tests:
--> main.o:
--> test_sender_main.o:
--> test_receiver_main.o:
--> test_subnet_main.o:
--> clean_simulator:
--> clean_tests:

you can update README file as things have changed now. you can refer to my project README file if you want.

you can probably rename main.cpp files for the sender, receiver, subnet. I know it is in separate folders but if someone by mistake swap those files, they won't know unless they open and see the contents. In this way, you can get rid of separate folders and keep all .cpp files in test/src folder itself rather than 3 sub-folders inside test/src.

Even this may (same name for all main.cpp files) introduces errors in the makefile. (professor discussed this thing in class and asked to rename main files)

**Comment by abhinavachanta** [comment]
Updated the code as per the review suggestions and committed the updated files. Pls review and approve

**Comment by syedomarazam** [comment]
completed code review for Task#1 after the developer fixed the bugs.

**Pull request merged**